packages/valory/skills/decision_maker_abci/behaviours/blacklisting.py

```diff
@@ -66,6 +66,14 @@ def async_act(self) -> Generator:
             yield from self.finish_behaviour(payload)
 
         with self.context.benchmark_tool.measure(self.behaviour_id).local():
+            if not self.policy:
+                success = yield from self._setup_policy_and_tools()
+                if not success:
+                    self.context.logger.info(
+                        "Tool selection failed, skipping blacklisting"
+                    )
+                    return
+
```

Comment:
The check is redundant as it's performed in the `_setup_policy_and_tools`.

```suggestion
            success = yield from self._setup_policy_and_tools()
            if not success:
                self.context.logger.info(
                    "Tool selection failed, skipping blacklisting"
                )
                return

```